platforms/software/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/TestsInitProjectSpecsViaPlugin.groovy

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins
+
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil
+import org.gradle.internal.jvm.Jvm
+import org.gradle.test.fixtures.plugin.PluginBuilder
+
+/**
+ * Fixture that provides for publishing a test plugin that adds a custom project spec to a
+ * local Maven repository for testing, and contain utilities for asserting that the plugin
+ * was resolved and its included spec was loaded.
+ */
```

Comment:
This will fix the unrecognized properties/methods below.

```suggestion
 */
 @SelfType(AbstractIntegrationSpec)
```

Reply:
I also renamed the type to be `TestsBuildInitSpecs...`